gridt/src/app/home/expandable/expandable.component.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { IonicModule } from '@ionic/angular';
+
+import { SignalMessageComponent } from './expandable.component';
+import { By } from '@angular/platform-browser';
+
+describe('ExpandableComponent', () => {
+  let component: SignalMessageComponent;
+  let fixture: ComponentFixture<SignalMessageComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ SignalMessageComponent ],
+      imports: [IonicModule.forRoot()]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SignalMessageComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  }));
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should toggle itemExpanded', () => {
+    component.signalText = "this message is long enough to require more space.";
+    component.width = 5;
```

Comment:
By setting the width yourself you are not checking the width determining code you have created. Is it possible to set the screen width in the test frame work?

Reply:
That is true! I hadn't thought of that. I will be keeping these tests i think and try and write 2 more to specifically test that if it can be set.

Reply:
So far, i haven't found a way to do this.

Reply:
I could expect width to be bigger than 0, that way we know the function in onInnit did something. But that doesn't help much as it could still be a wrong value.